angr/engines/concrete.py

```diff
@@ -0,0 +1,168 @@
+import logging
+import signal
+import sys
+
+from .engine import SimEngine
+from ..errors import SimConcreteMemoryError, SimConcreteRegisterError
+
+l = logging.getLogger("angr.engines.concrete")
+
+try:
+    from angr_targets.concrete import ConcreteTarget
+except ImportError as e:
+    l.error("Can't find angr_targets project!")
+
+
+def timeout_handler():
+    l.critical("[ERROR] Timeout error during resuming of concrete process")
+    # situation is compromised, better to exit.
+    sys.exit()
```

Comment:
I wonder what the logic of timeout is -- are we forcing angr to quit if the concrete target times out? This sounds like a weird choice to me. Under what circumstances will this be triggered?